src/Interfaces/AbstractPolytope.jl

```diff
@@ -144,13 +144,12 @@ end
     end
 end
 
-@inline function _linear_map_hrep(M::AbstractMatrix{N}, P::AbstractPolytope{N},
-                                  use_inv::Bool;
-                                  inverse::Union{Nothing, AbstractMatrix{N}}=nothing
-                                 ) where {N<:Real}
-    constraints = _linear_map_hrep_helper(M, P, use_inv; inverse=inverse)
+function _linear_map_hrep_helper(M::AbstractMatrix{N}, P::AbstractPolytope{N},
+                                 algo::AbstractLinearMapAlgorithm) where {N<:Real}
+    constraints = _linear_map_hrep(M, P, algo)
     m = size(M, 1) # output dimension
     if m == 1
+        # TODO: create interval directly ?
         return convert(Interval, HPolygon(constraints))
```

Comment:
Good point. We can solve that later.
How would it look like? Do we know that `_linear_map_hrep` returns exactly two constraints?

Reply:
> How would it look like?
> Do we know that _linear_map_hrep returns exactly two constraints?

I can't guarantee it always returns 2 constraints.. as we're not explicitly removing redundant constraints..

Reply:
Now that I see this again, `convert(Interval, HPolygon)` looks really wrong to me. Why does this even work? :smile:

Reply:
:smile:  it was silly to convert from a polygon to an interval, it doesn't work (and it shouldn't):

```julia
julia> convert(Interval, rand(HPolygon))
ERROR: AssertionError: can only convert a one-dimensional HPolygon{Float64} to `Int
erval`
Stacktrace:
 [1] macro expansion at /home/mforets/.julia/dev/LazySets/src/Assertions/Assertions
.jl:23 [inlined]
 [2] convert(::Type{Interval}, ::HPolygon{Float64}) at /home/mforets/.julia/dev/Laz
ySets/src/convert.jl:513
 [3] top-level scope at REPL[6]:1
```
So i propose to change that line to `convert(Interval, HPolytope(constraints))`.